test/testGetMetaPair.js

```diff
@@ -0,0 +1,231 @@
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'POST', _tokendata:{ userid: 'sally', isserver: true }, params:{userid:'sally'}};
+
+      expect(getMeta(req,res, done));
+    });
+    it('should not worry that the userid differ', function () {
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'GET', _tokendata:{ userid: 'sally', isserver: true }, params:{userid:'billy'}};
+
+      expect(getMeta(req,res, done));
+    });
+  });
+
+  describe('given a user token for requested userid', function () {
+    it('should be valid when id matches', function () {
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'GET', _tokendata:{ userid: 'sally', isserver: false }, params:{userid:'sally'}};
+
+      expect(getMeta(req, res, done));
+    });
+    it('should be valid for PUT method', function () {
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'PUT', _tokendata:{ userid: 'sally', isserver: false }, params:{userid:'sally'}};
+
+      expect(getMeta(req, res, done));
+    });
+    it('should be valid for POST method', function () {
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'POST', _tokendata:{ userid: 'sally', isserver: false }, params:{userid:'sally'}};
+
+      expect(getMeta(req, res, done));
+    });
+    it('should be valid for DEL method', function () {
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'DEL', _tokendata:{ userid: 'sally', isserver: false }, params:{userid:'sally'}};
+
+      expect(getMeta(req, res, done));
+    });
+  });
+
+  describe('given a user token for different userid', function () {
+    it('should invalid when not in group', function () {
+      setupStubs(emptyResponse, emptyGroups);
+
+      var done = function(result){
+        expect(result).to.be.false;
+      };
+
+      var req = { method:'GET', _tokendata:{ userid: 'sally', isserver: false }, params:{userid:'other'}};
+
+      expect(getMeta(req,res, done));
+    });
+    it('should be valid when in group and GET', function () {
+      setupStubs(emptyResponse, inGroups);
+
+      var done = function(result){
+        expect(result).to.be.empty;
+      };
+
+      var req = { method:'GET', _tokendata:{ userid: 'sally', isserver: false }, params:{userid:'other'}};
+
+      expect(getMeta(req, res, done));
+    });
+    it('should be valid when in group and POST', function () {
```

Comment:
I thought we were going to only allow read (GET) for a user in the group (ie. not the user themselves).